File: bank-service/src/main/java/rs/edu/raf/model/accounts/ForeignCurrencyBankAccount.java
package rs.edu.raf.model.accounts;


import jakarta.persistence.*;


import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.math.BigDecimal;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class ForeignCurrencyBankAccount extends BankAccount {
    private BigDecimal maintenancePrice;
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/model/accounts/ForeignCurrencyBankAccount.kt
package org.example.bankservicekotlin.model.accounts

import jakarta.persistence.Entity
import org.example.bankservicekotlin.model.Currency
import java.math.BigDecimal

@Entity
class ForeignCurrencyBankAccount(
    id: Long?,
    accountNumber: Long?,
    owner: Long?,
    balance: BigDecimal,
    availableBalance: BigDecimal,
    creator: Long, // employee who created account
    creationDate: Long,
    currency: Currency,
    active: Boolean,
    val maintenancePrice: BigDecimal
) : BankAccount(id, accountNumber, owner, balance, availableBalance, creator, creationDate, currency, active) {
}